plugins/GeoIp2/LocationProvider/GeoIp2/Php.php

```diff
@@ -0,0 +1,288 @@
+                        break;
+                }
+            } catch (AddressNotFoundException $e) {
+                // ignore - do nothing
+            }
+        }
+
+        // NOTE: ISP & ORG require commercial dbs to test.
+        $ispGeoIp = $this->getGeoIpInstance($key = 'isp');
+        if ($ispGeoIp) {
+            try {
+                $lookupResult          = $ispGeoIp->isp($ip);
+                $result[self::ISP_KEY] = $lookupResult->isp;
+                $result[self::ORG_KEY] = $lookupResult->organization;
+            } catch (AddressNotFoundException $e) {
+                // ignore - do nothing
+            }
+        }
+
+        if (empty($result)) {
+            return false;
+        }
+
+        $this->completeLocationResult($result);
+        return $result;
+    }
+
+    /**
+     * Returns true if this location provider is available. Piwik ships w/ the MaxMind
+     * PHP library, so this provider is available if a location GeoIP database can be found.
+     *
+     * @return bool
+     */
+    public function isAvailable()
+    {
+        $path = self::getPathToGeoIpDatabase($this->customDbNames['loc']);
+        return $path !== false;
+    }
+
+    /**
+     * Returns an array describing the types of location information this provider will
+     * return.
+     *
+     * The location info this provider supports depends on what GeoIP databases it can
+     * find.
+     *
+     * This provider will always support country & continent information.
+     *
+     * If a region database is found, then region code & name information will be
+     * supported.
+     *
+     * If a city database is found, then region code, region name, city name,
+     * area code, latitude, longitude & postal code are all supported.
+     *
+     * If an ISP/organization database is found, ISP/organization information is supported.
+     *
+     * @return array
+     */
+    public function getSupportedLocationInfo()
+    {
+        $result = array();
+
+        // country & continent info always available
+        $result[self::CONTINENT_CODE_KEY] = true;
+        $result[self::CONTINENT_NAME_KEY] = true;
+        $result[self::COUNTRY_CODE_KEY] = true;
+        $result[self::COUNTRY_NAME_KEY] = true;
+
+        $reader = $this->getGeoIpInstance($key = 'loc');
+        if ($reader) {
+            switch ($reader->metadata()->databaseType) {
+                case 'GeoIP2-Enterprise':
+                case 'GeoIP2-City':
+                case 'GeoLite2-City':
+                    $result[self::REGION_CODE_KEY] = true;
+                    $result[self::REGION_NAME_KEY] = true;
+                    $result[self::CITY_NAME_KEY] = true;
+                    $result[self::POSTAL_CODE_KEY] = true;
+                    $result[self::LATITUDE_KEY] = true; // will be removed from GeoLite2-City in 2019
+                    $result[self::LONGITUDE_KEY] = true; // will be removed from GeoLite2-City in 2019
```

Comment:
It will only be removed from GeoLite database published in 2019 and the paid versions will still contain such data.
Also if someone is using an old database and doesn't update after 2019 he still would have those values... Haven't spend much thought on that, but we have [an issue](https://github.com/matomo-org/matomo/issues/12411) for that. Aldo there might be some changes required for tracking lat/long at all re GDPR, so guess no need to add some checks in code for now.

Reply:
Gotcha